model/change_point_detection.go

```diff
@@ -369,10 +369,9 @@ func ReplaceChangePoints(ctx context.Context, env cedar.Environment, performance
 	return catcher.Resolve()
 }
 
-func GetTotalPagesForChangePointsGroupedByVersion(ctx context.Context, env cedar.Environment, projectId string, pageSize int) (int, error) {
-	countKey := "count"
-	pipe := appendAfterBaseGetChangePointsByVersionAgg(projectId, bson.M{
-		"$count": countKey,
+func GetTotalPagesForChangePointsGroupedByVersion(ctx context.Context, env cedar.Environment, projectId string, pageSize int, variantRegex, versionRegex, taskRegex, testRegex, measurementRegex string, threadLevels []int) (int, error) {
```

Comment:
lets use an arg struct here

Reply:
Which package/s should this arg struct reside in?

It seems like this coerces either:
* mapping over structs in each layer (inefficient)
* direct dependency of top level packages on bottom level packages (design antipattern)

Reply:
The arg struct would reside in the model package. What do you mean by mapping over struct in each layer?

Reply:
Here's an example of an arg struct: https://github.com/evergreen-ci/cedar/blob/master/rest/data/interface.go#L97

Reply:
If the arg struct resides in the model package, then the rest package needs to know about that package, and that defeats the purpose of providing interfaces in the data package that are meant to abstract away the data persistence layer.

Reply:
if you add an arg struct in the data package to allow rest package to be agnostic of the model package, then you will need to map over from one arg struct to another.

Reply:
Regarding indexes, there appear to be some in place for project, variant, version, task, test.

Reply:
The rest/data package already uses the model package so it doesn't matter. You can either map over from one arg struct to another, or you can alias the model's arg struct in the rest/data package so that the rest package is still agnostic to the model package.

Reply:
I'd argue the rest package should be refactored at some point to *not* use the model package directly. In any case, I didn't know about type aliases, and that solves the problem!